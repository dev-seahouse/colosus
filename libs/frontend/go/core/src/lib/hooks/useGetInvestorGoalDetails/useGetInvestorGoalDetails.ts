import type {
  InvestorGetGoalDetailsApiRequestDto,
  InvestorGetGoalDetailsApiResponseDto,
} from '@bambu/api-client';
import { TransactInvestorAuthenticatedGoalsApi } from '@bambu/api-client';
import { QueryClient, useQuery } from '@tanstack/react-query';
import { QueryArgs } from '../../types/utils';

const getInvestorGoalDetailsQuery = (
  args: InvestorGetGoalDetailsApiRequestDto
) => ({
  queryKey: ['getInvestorGoalDetails', args],
  queryFn: () => fetchInvestorGoalDetails(args),
});

export function useGetInvestorGoalDetails(
  args: InvestorGetGoalDetailsApiRequestDto,
  queryOptions?: QueryArgs<InvestorGetGoalDetailsApiResponseDto>
) {
  return useQuery({
    ...getInvestorGoalDetailsQuery(args),
    ...queryOptions,
  });
}

export const getInvestorGoalDetailsLoader = (
  queryClient: QueryClient,
  params: InvestorGetGoalDetailsApiRequestDto
) => {
  return async () => {
    const query = getInvestorGoalDetailsQuery(params);
    return (
      queryClient.getQueryData(query.queryKey) ??
      (await queryClient.fetchQuery(query))
    );
  };
};

async function fetchInvestorGoalDetails(
  args: InvestorGetGoalDetailsApiRequestDto
) {
  const goalsApi = new TransactInvestorAuthenticatedGoalsApi();
  const res = await goalsApi.getInvestorGoalDetails(args);
  return res.data;
}
export default useGetInvestorGoalDetails;
