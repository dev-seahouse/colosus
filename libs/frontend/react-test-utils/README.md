# BAMBU React Test Utils

This library contains modified version of `@testing-library/react` render function and re-exported other utilities functions from `@testing-library/react` and `@testing-library/user-event`.
