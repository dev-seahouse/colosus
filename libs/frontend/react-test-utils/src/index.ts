export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render } from './lib/customRender/customRender';
export { renderWithDataRouter } from './lib/renderWithDataRouter/renderWithDataRouter';
export * from './lib/queryClientWrapper/queryClientWrapper';
export * from './lib/reactRouterWrapper/reactRouterWrapper';

export * from './lib/ReactHookFormWrapper/ReactHookFormWrapper';
