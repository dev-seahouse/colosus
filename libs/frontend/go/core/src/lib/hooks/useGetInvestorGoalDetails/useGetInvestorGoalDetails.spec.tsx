import { renderHook, waitFor } from '@testing-library/react';

import useGetInvestorGoalDetails from './useGetInvestorGoalDetails';
import { queryClientWrapper } from '@bambu/react-test-utils';

describe('useGetInvestorGoalDetails', () => {
  it('should render successfully', async () => {
    const { result } = renderHook(
      () => useGetInvestorGoalDetails({ goalId: 'dfdfd' }),
      { wrapper: queryClientWrapper }
    );
    await waitFor(() => expect(result.current.isSuccess).toBe(true));
    expect(result.current.data).toMatchSnapshot();
  });
});
