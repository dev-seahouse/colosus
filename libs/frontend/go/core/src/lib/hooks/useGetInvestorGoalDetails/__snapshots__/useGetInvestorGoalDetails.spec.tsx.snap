// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGetInvestorGoalDetails > should render successfully 1`] = `
{
  "ConnectPortfolioSummary": {
    "RiskProfile": {
      "createdAt": "2023-09-06T07:40:23.134Z",
      "createdBy": "COLOSSUS MIGRATION SCRIPT",
      "id": "d30d1d4b-97aa-4028-9e56-c651004a106c",
      "lowerLimit": "5",
      "riskProfileDescription": "You are OK with high volatility in your portfolio.<br/>You understand that the value of your portfolio may go down sharply in the future but you know that you will reap big benefits if you are patient enough.<br/>You expect high returns in the long term.",
      "riskProfileName": "Aggressive",
      "tenantId": "c1f400b5-6aeb-48fa-a38e-c1f30cf716de",
      "updatedAt": "2023-10-10T03:13:10.748Z",
      "updatedBy": "COLOSSUS MIGRATION SCRIPT",
      "upperLimit": "5",
    },
    "TransactModelPortfolios": [
      {
        "TransactModelPortfolioInstruments": [
          {
            "Instrument": {
              "InstrumentAssetClass": {
                "createdAt": "2023-10-19T13:51:04.343Z",
                "createdBy": "QUARK",
                "id": "63af8e9c-c378-4068-b96e-0e63ca9d629e",
                "name": "Equities",
                "updatedAt": "2023-10-19T13:51:04.343Z",
                "updatedBy": "QUARK",
              },
              "InstrumentCurrency": {
                "createdAt": "2023-10-19T13:44:15.035Z",
                "createdBy": "QUARK",
                "id": "c6e977a6-5208-4726-bdf5-e905ee11a7ff",
                "iso4217Code": "GBP",
                "updatedAt": "2023-10-19T13:44:15.035Z",
                "updatedBy": "QUARK",
              },
              "InstrumentExchange": {
                "bambuExchangeCode": "XLON",
                "createdAt": "2023-10-19T13:48:04.826Z",
                "createdBy": "QUARK",
                "id": "9743a4b3-fcbc-4bcd-9200-d64b184d7619",
                "updatedAt": "2023-10-19T13:48:04.826Z",
                "updatedBy": "QUARK",
              },
              "InstrumentFactSheets": [
                {
                  "createdAt": "2023-10-19T14:04:33.122Z",
                  "createdBy": "QUARK",
                  "id": "b538789e-58ee-4dbc-a623-131882fdc582",
                  "instrumentId": "149696a7a208d8f3074ee563f7a42d5b",
                  "updatedAt": "2023-10-19T14:04:33.122Z",
                  "updatedBy": "QUARK",
                  "url": "https://www.ishares.com/uk/individual/en/products/251795/",
                },
              ],
              "bloombergTicker": "ISF",
              "createdAt": "2023-10-19T14:04:33.119Z",
              "createdBy": "QUARK",
              "cusip": null,
              "id": "149696a7a208d8f3074ee563f7a42d5b",
              "instrumentAssetClassId": "63af8e9c-c378-4068-b96e-0e63ca9d629e",
              "instrumentCurrencyId": "c6e977a6-5208-4726-bdf5-e905ee11a7ff",
              "instrumentExchangeId": "9743a4b3-fcbc-4bcd-9200-d64b184d7619",
              "isin": "IE0005042456",
              "name": "iShares Core FTSE 100 UCITS ETF GBP Dist",
              "ricSymbol": null,
              "updatedAt": "2023-10-26T02:02:27.737Z",
              "updatedBy": "QUARK",
            },
            "createdAt": "2023-10-24T17:56:57.192Z",
            "createdBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "id": "bb4ee6a2-29ab-4545-86a2-df60eda60597",
            "instrumentId": "149696a7a208d8f3074ee563f7a42d5b",
            "transactModelPortfolioId": "05ed5255-4e1f-4324-bee3-cea1a7797d6e",
            "updatedAt": "2023-10-24T17:56:57.192Z",
            "updatedBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "weightage": 0.28,
          },
          {
            "Instrument": {
              "InstrumentAssetClass": {
                "createdAt": "2023-10-19T13:51:04.343Z",
                "createdBy": "QUARK",
                "id": "63af8e9c-c378-4068-b96e-0e63ca9d629e",
                "name": "Equities",
                "updatedAt": "2023-10-19T13:51:04.343Z",
                "updatedBy": "QUARK",
              },
              "InstrumentCurrency": {
                "createdAt": "2023-10-19T13:44:15.035Z",
                "createdBy": "QUARK",
                "id": "e5b5eab9-3e9c-443d-88b5-db3292176802",
                "iso4217Code": "USD",
                "updatedAt": "2023-10-19T13:44:15.035Z",
                "updatedBy": "QUARK",
              },
              "InstrumentExchange": {
                "bambuExchangeCode": "XLON",
                "createdAt": "2023-10-19T13:48:04.826Z",
                "createdBy": "QUARK",
                "id": "9743a4b3-fcbc-4bcd-9200-d64b184d7619",
                "updatedAt": "2023-10-19T13:48:04.826Z",
                "updatedBy": "QUARK",
              },
              "InstrumentFactSheets": [
                {
                  "createdAt": "2023-10-19T14:04:33.059Z",
                  "createdBy": "QUARK",
                  "id": "4e24ac77-952d-4bcd-8ba4-2baa85e230a8",
                  "instrumentId": "597bb65ddccc3503df81d2283b95d1d9",
                  "updatedAt": "2023-10-19T14:04:33.059Z",
                  "updatedBy": "QUARK",
                  "url": "https://www.ishares.com/uk/individual/en/products/251882/ishares-msci-world-ucits-etf-acc-fund",
                },
              ],
              "bloombergTicker": "IWDA",
              "createdAt": "2023-10-19T14:04:33.049Z",
              "createdBy": "QUARK",
              "cusip": null,
              "id": "597bb65ddccc3503df81d2283b95d1d9",
              "instrumentAssetClassId": "63af8e9c-c378-4068-b96e-0e63ca9d629e",
              "instrumentCurrencyId": "e5b5eab9-3e9c-443d-88b5-db3292176802",
              "instrumentExchangeId": "9743a4b3-fcbc-4bcd-9200-d64b184d7619",
              "isin": "IE00B4L5Y983",
              "name": "iShares Core MSCI World UCITS ETF USD (Acc)",
              "ricSymbol": null,
              "updatedAt": "2023-10-26T02:02:27.696Z",
              "updatedBy": "QUARK",
            },
            "createdAt": "2023-10-24T17:56:57.191Z",
            "createdBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "id": "967eebfe-bd8f-49ce-9277-9245f3cd1cd3",
            "instrumentId": "597bb65ddccc3503df81d2283b95d1d9",
            "transactModelPortfolioId": "05ed5255-4e1f-4324-bee3-cea1a7797d6e",
            "updatedAt": "2023-10-24T17:56:57.191Z",
            "updatedBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "weightage": 0.5,
          },
          {
            "Instrument": {
              "InstrumentAssetClass": {
                "createdAt": "2023-10-19T13:51:04.343Z",
                "createdBy": "QUARK",
                "id": "63af8e9c-c378-4068-b96e-0e63ca9d629e",
                "name": "Equities",
                "updatedAt": "2023-10-19T13:51:04.343Z",
                "updatedBy": "QUARK",
              },
              "InstrumentCurrency": {
                "createdAt": "2023-10-19T13:44:15.035Z",
                "createdBy": "QUARK",
                "id": "e5b5eab9-3e9c-443d-88b5-db3292176802",
                "iso4217Code": "USD",
                "updatedAt": "2023-10-19T13:44:15.035Z",
                "updatedBy": "QUARK",
              },
              "InstrumentExchange": {
                "bambuExchangeCode": "XLON",
                "createdAt": "2023-10-19T13:48:04.826Z",
                "createdBy": "QUARK",
                "id": "9743a4b3-fcbc-4bcd-9200-d64b184d7619",
                "updatedAt": "2023-10-19T13:48:04.826Z",
                "updatedBy": "QUARK",
              },
              "InstrumentFactSheets": [
                {
                  "createdAt": "2023-10-19T14:04:33.306Z",
                  "createdBy": "QUARK",
                  "id": "8d2fb9dc-9d5f-4724-a28c-54ae6766e48d",
                  "instrumentId": "912e7f76cfb94307876bcfc3d2bdcf28",
                  "updatedAt": "2023-10-19T14:04:33.306Z",
                  "updatedBy": "QUARK",
                  "url": "https://www.ishares.com/uk/individual/en/products/296576/?referrer=tickerSearch",
                },
              ],
              "bloombergTicker": "WSML",
              "createdAt": "2023-10-19T14:04:33.305Z",
              "createdBy": "QUARK",
              "cusip": null,
              "id": "912e7f76cfb94307876bcfc3d2bdcf28",
              "instrumentAssetClassId": "63af8e9c-c378-4068-b96e-0e63ca9d629e",
              "instrumentCurrencyId": "e5b5eab9-3e9c-443d-88b5-db3292176802",
              "instrumentExchangeId": "9743a4b3-fcbc-4bcd-9200-d64b184d7619",
              "isin": "IE00BF4RFH31",
              "name": "iShares MSCI World Small Cap UCITS ETF USD (Acc)",
              "ricSymbol": null,
              "updatedAt": "2023-10-26T02:02:27.848Z",
              "updatedBy": "QUARK",
            },
            "createdAt": "2023-10-26T01:17:25.000Z",
            "createdBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "id": "79d6dad0-735a-11ee-b962-0242ac120002",
            "instrumentId": "912e7f76cfb94307876bcfc3d2bdcf28",
            "transactModelPortfolioId": "05ed5255-4e1f-4324-bee3-cea1a7797d6e",
            "updatedAt": "2023-10-26T01:17:26.000Z",
            "updatedBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "weightage": 0.2,
          },
          {
            "Instrument": {
              "InstrumentAssetClass": {
                "createdAt": "2023-10-25T17:13:00.790Z",
                "createdBy": "QUARK",
                "id": "ae5c977d-bd68-4d8a-82b0-3e778bbb5a4c",
                "name": "Cash",
                "updatedAt": "2023-10-25T17:13:00.790Z",
                "updatedBy": "QUARK",
              },
              "InstrumentCurrency": {
                "createdAt": "2023-10-19T13:44:15.035Z",
                "createdBy": "QUARK",
                "id": "c6e977a6-5208-4726-bdf5-e905ee11a7ff",
                "iso4217Code": "GBP",
                "updatedAt": "2023-10-19T13:44:15.035Z",
                "updatedBy": "QUARK",
              },
              "InstrumentExchange": {
                "bambuExchangeCode": "CASH",
                "createdAt": "2023-10-26T02:02:27.665Z",
                "createdBy": "QUARK",
                "id": "b3470107-8033-46f0-bf3a-66fa81897855",
                "updatedAt": "2023-10-26T02:02:27.665Z",
                "updatedBy": "QUARK",
              },
              "InstrumentFactSheets": [],
              "bloombergTicker": "CASH_GBP",
              "createdAt": "2023-10-26T02:02:27.873Z",
              "createdBy": "QUARK",
              "cusip": null,
              "id": "c17f9b17d0d8177afae3453429e221aa",
              "instrumentAssetClassId": "ae5c977d-bd68-4d8a-82b0-3e778bbb5a4c",
              "instrumentCurrencyId": "c6e977a6-5208-4726-bdf5-e905ee11a7ff",
              "instrumentExchangeId": "b3470107-8033-46f0-bf3a-66fa81897855",
              "isin": "CASH_GBP",
              "name": "Cash - Pound Sterling",
              "ricSymbol": null,
              "updatedAt": "2023-10-26T02:02:27.873Z",
              "updatedBy": "QUARK",
            },
            "createdAt": "2023-10-26T01:17:25.000Z",
            "createdBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "id": "79d6dd64-735a-11ee-b962-0242ac120002",
            "instrumentId": "c17f9b17d0d8177afae3453429e221aa",
            "transactModelPortfolioId": "05ed5255-4e1f-4324-bee3-cea1a7797d6e",
            "updatedAt": "2023-10-26T01:17:28.000Z",
            "updatedBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
            "weightage": 0.02,
          },
        ],
        "connectPortfolioSummaryId": "05ed5255-4e1f-4324-bee3-cea1a7797d6e",
        "createdAt": "2023-10-24T16:11:16.475Z",
        "createdBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
        "description": "string",
        "expectedAnnualReturn": null,
        "expectedAnnualVolatility": null,
        "factSheetUrl": "string",
        "id": "05ed5255-4e1f-4324-bee3-cea1a7797d6e",
        "name": "string",
        "partnerModelId": "mdl-36h4zk4q6242wa",
        "rebalancingThreshold": null,
        "updatedAt": "2023-10-26T02:20:24.688Z",
        "updatedBy": "c1631f0f-50b0-437c-b728-1fb9f29b74b6",
      },
    ],
    "assetClassAllocation": [
      {
        "assetClass": "Equity",
        "included": true,
        "percentOfPortfolio": "100",
      },
      {
        "assetClass": "Money Market",
        "included": true,
        "percentOfPortfolio": "0",
      },
      {
        "assetClass": "Bonds",
        "included": true,
        "percentOfPortfolio": "0",
      },
      {
        "assetClass": "Other",
        "included": true,
        "percentOfPortfolio": "0",
      },
    ],
    "description": "This portfolio is suitable for investors seeking high returns and who are willing to take on high risk. It has a long-term investment time horizon.",
    "expectedReturnPercent": "8",
    "expectedVolatilityPercent": "18",
    "id": "05ed5255-4e1f-4324-bee3-cea1a7797d6e",
    "key": "AGGRESSIVE",
    "name": "Aggressive Portfolio",
    "reviewed": false,
    "risk_profile_id": "d30d1d4b-97aa-4028-9e56-c651004a106c",
    "showSummaryStatistics": true,
    "sortKey": 50,
    "tenantId": "c1f400b5-6aeb-48fa-a38e-c1f30cf716de",
  },
  "GoalRecurringSavingsPlans": [
    {
      "amount": 1000,
      "createdAt": "2023-10-21T18:59:39.122Z",
      "createdBy": "COLOSSUS",
      "currency": "GBP",
      "data": {
        "projectedReturns": {
          "high": 0,
          "low": 0,
          "target": 0,
        },
      },
      "frequency": "MONTHLY",
      "goalId": "0a9cade0-66ec-479b-b6f3-ad7c3fafb55f",
      "id": "43cfbf67-fd96-4097-b4ba-013086e6cd74",
      "status": "PENDING",
      "updatedAt": "2023-10-21T18:59:39.122Z",
      "updatedBy": "COLOSSUS",
    },
  ],
  "computedRiskProfile": {
    "riskAppetite": "bca96485-8dc2-41fc-aabf-2a3f4745aab8",
    "riskProfileId": "4bf6874c-e5ea-4f66-b2f9-e1212b823422",
  },
  "connectPortfolioSummaryId": "bca96485-8dc2-41fc-aabf-2a3f4745aab8",
  "createdAt": "2023-10-21T18:59:39.122Z",
  "createdBy": "COLOSSUS",
  "data": {
    "projectedReturns": {
      "high": 0,
      "low": 0,
      "target": 0,
    },
  },
  "goalDescription": "Buy a house",
  "goalEndDate": null,
  "goalName": "House",
  "goalStartDate": null,
  "goalTimeframe": 4,
  "goalValue": 60000,
  "id": "0a9cade0-66ec-479b-b6f3-ad7c3fafb55f",
  "initialInvestment": 20000,
  "investorId": "e25a987f-25b8-4af1-b7e5-bd9547a721b8",
  "portfolioValue": 0,
  "portfolioValueCurrency": "GBP",
  "portfolioValueDate": "2023-10-26",
  "sendLeadAppointmentEmail": false,
  "sendLeadGoalProjectionEmail": false,
  "status": "PENDING",
  "updatedAt": "2023-10-21T18:59:39.122Z",
  "updatedBy": "COLOSSUS",
}
`;
